lib/growthbook/util.rb

```diff
@@ -62,12 +62,15 @@ def self.chooseVariation(userId, experiment)
       match
     end
 
-    def self.hash(str)
-      (FNV.new.fnv1a_32(str) % 1000) / 1000.0
+    def self.hash(seed:, value:, version:)
+      return (FNV.new.fnv1a_32(value + seed) % 1000) / 1000.0 if version == 1
+      return (FNV.new.fnv1a_32(FNV.new.fnv1a_32(seed + value).to_s) % 10000) / 10000.0 if version == 2
+
+      -1
```

Comment:
Based on the updated 0.4.1 tests, this method should return nil. We are doing a less than check which returns truthy for -1 when it shouldn't, and this was fixed in 0.4.1.